inst/tests/tests.Rraw

```diff
@@ -18748,3 +18748,11 @@ test(2270, options=c(datatable.optimize=1L), DT[, mean(b, 1), by=a], data.table(
 DT1 = data.table(a=1:2)
 DT2 = data.table(a=c(1, 1, 2, 2), b=1:4)
 test(2271, options=c(datatable.verbose=TRUE), copy(DT1)[DT2, on='a', v := 4], copy(DT1)[, v := 4], output="x.a.\nAssigning")
+
+# convert POSIXlt to POSIXct in set and := #1724
+dt = data.table(a=1:2)
+dates = as.POSIXlt(c("2024-01-01", "2024-01-02"), tz="")
+ans = data.table(a=1:2, dates=as.POSIXct(dates))
+test(2272.1, copy(dt)[, dates := dates], ans, warning="POSIXlt.*converted to POSIXct")
+test(2272.2, set(copy(dt), j="dates", value=dates), ans, warning="POSIXlt.*converted to POSIXct")
+test(2272.3, {dt1 = copy(dt); dt1$dates=dates; dt1}, ans, warning="POSIXlt.*converted to POSIXct")
```

Comment:
what about a test like `DT[, new_var := strptime(something)]` ?
I would think that the code in this PR makes the existing code and warning (below) obsolete, what do you think?

strptime() usage detected and wrapped with as.POSIXct(). This is to minimize the chance of assigning POSIXlt columns, which use 40+ bytes to store one date (versus 8 for POSIXct). Use as.POSIXct() (which will call strptime() as needed internally) to avoid this warning.

Reply:
For the assignemnt part yes, but you do not have to assign in j, hence, we still need it for `DT[, strptime(something)]`

Reply:
do we need it? the strptime warning says "this is to minimize the chance of assigning POSIXlt columns" but that should no longer be possible after this PR, right?

Reply:
True that.